client/packages/system/src/Item/Components/PackVariantEditModal/PackVariantEditModal.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { FC, useState } from 'react';
+import {
+  ModalMode,
+  useDialog,
+  BasicTextInput,
+  Grid,
+  DialogButton,
+  useTranslation,
+  FnUtils,
+  InlineSpinner,
+  NonNegativeIntegerInput,
+  useNotification,
+  noOtherVariants,
+} from '@openmsupply-client/common';
+import { VariantFragment } from '../../api';
+import { usePackVariantInsert } from '../../api/hooks/usePackVariantInsert';
+import { usePackVariantUpdate } from '../../api/hooks/usePackVariantUpdate';
+
+interface PackVariantEditModalProps {
+  itemId: string;
+  mode: ModalMode | null;
+  isOpen: boolean;
+  onClose: () => void;
+  packVariant: VariantFragment | null;
+}
+
+const createNewPackVariant = (itemId: string): VariantFragment => ({
+  __typename: 'VariantNode',
+  id: FnUtils.generateUUID(),
+  itemId,
+  shortName: '',
+  longName: '',
+  packSize: 0,
+});
+
+interface UseDraftPackVariantControl {
+  draft: VariantFragment;
+  onUpdate: (patch: Partial<VariantFragment>) => void;
+  onSave: () => Promise<void>;
+  isLoading: boolean;
+}
+
+const useDraftPackVariant = (
+  itemId: string,
+  seed: VariantFragment | null,
+  mode: ModalMode | null
+): UseDraftPackVariantControl => {
+  const [packVariant, setPackVariant] = useState<VariantFragment>(
+    () => seed || createNewPackVariant(itemId)
+  );
+  const t = useTranslation('catalogue');
+  const { error } = useNotification();
+  const { mutateAsync: insert, isLoading: insertIsLoading } =
+    usePackVariantInsert();
+  const { mutateAsync: update, isLoading: updateIsLoading } =
+    usePackVariantUpdate();
+
+  const onUpdate = (patch: Partial<VariantFragment>) => {
+    setPackVariant({ ...packVariant, ...patch });
+  };
+
+  const onSave = async () => {
+    let result =
+      mode === ModalMode.Create
+        ? await insert(packVariant)
+        : await update(packVariant);
+
+    if (result.__typename === 'VariantNode') return;
+    const structuredError = result.error;
+
+    switch (structuredError.__typename) {
+      case 'VariantWithPackSizeAlreadyExists':
+        error(t('error.pack-variant-exists'))();
+        throw Error();
+      default:
+        noOtherVariants(structuredError.__typename);
+    }
+  };
+
+  return {
+    draft: packVariant,
+    onUpdate,
+    onSave,
+    isLoading: insertIsLoading || updateIsLoading,
+  };
+};
+
+export const PackVariantEditModal: FC<PackVariantEditModalProps> = ({
+  itemId,
+  mode,
+  isOpen,
+  onClose,
+  packVariant,
+}) => {
+  const { Modal } = useDialog({ isOpen, onClose });
+  const t = useTranslation('catalogue');
+  const { draft, onUpdate, onSave, isLoading } = useDraftPackVariant(
+    itemId,
+    packVariant,
+    mode
+  );
+  const isInvalid = !draft.packSize && !draft.itemId;
+
+  return (
+    <Modal
```

Comment:
The modal is pretty wide, and the full-width inputs look strange to me, particularly the numeric input.
Would you mind:

<img width="400" alt="Screenshot 2023-11-22 at 10 59 29 AM" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/3cf2bd7e-58a1-413e-b185-c641c5f31c8c">

or right aligned?

<img width="410" alt="Screenshot 2023-11-22 at 11 03 14 AM" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/20bbb665-f4ef-49fe-8d8e-6f22c37172fe">

Reply:
I thought that looked weird... So I made it full width but will change

Reply:
yep. it's a bit weird - personally, I prefer the left-aligned small width pack size input version.. it's very subjective though